cmd/gorepomod/internal/utils/utils.go

```diff
@@ -45,3 +47,38 @@ func SliceContains(slice []string, target string) bool {
 	}
 	return false
 }
+
+// Receive git remote url as input and produce string containing git repository name
+// e.g. kustomize, myrepo/kustomize
+func ParseGitRepositoryPath(urlString string) string {
+	var d []string = getURLStringArray(urlString)
+	protocol := d[0]
+
+	var repoPath string
+
+	// for ssh protocol git@github.com:path/repo.git
+	switch protocol {
+	case "git":
+		repoPath = strings.Join(d[2:len(d)-1], "/") + "/" + d[3][:len(d[3])-4]
+		// https protocol https://github.com/path/repo.git
+	case "https":
+		repoPath = strings.Join(d[2:], "/")
+		repoPath = repoPath[:len(repoPath)-4]
+		// unsupported format
+		// TODO: confirm if we should handle other formats not commonly supported by Github
```

Comment:
Please write your GitHub id in the TODO comment.
https://github.com/kubernetes-sigs/kustomize/blob/0b4974eb1ce0316c7b1c2f82365c858b5c9b37d4/kyaml/openapi/openapi.go#L675

This ID is super useful when tracking TODO.